account_asset_management/wizard/account_asset_remove.py

```diff
@@ -0,0 +1,331 @@
+        default=fields.Date.today,
+        help="Removal date must be after the last posted entry "
+             "in case of early removal")
+    force_date = fields.Date(
+        string='Force accounting date')
+    sale_value = fields.Float(
+        string='Sale Value',
+        default=lambda self: self._default_sale_value())
+    account_sale_id = fields.Many2one(
+        comodel_name='account.account',
+        string='Asset Sale Account',
+        domain=[('deprecated', '=', False)],
+        default=lambda self: self._default_account_sale_id())
+    account_plus_value_id = fields.Many2one(
+        comodel_name='account.account',
+        string='Plus-Value Account',
+        domain=[('deprecated', '=', False)],
+        default=lambda self: self._default_account_plus_value_id())
+    account_min_value_id = fields.Many2one(
+        comodel_name='account.account',
+        string='Min-Value Account',
+        domain=[('deprecated', '=', False)],
+        default=lambda self: self._default_account_min_value_id())
+    account_residual_value_id = fields.Many2one(
+        comodel_name='account.account',
+        string='Residual Value Account',
+        domain=[('deprecated', '=', False)],
+        default=lambda self: self._default_account_residual_value_id())
+    posting_regime = fields.Selection(
+        selection=lambda self: self._selection_posting_regime(),
+        string='Removal Entry Policy',
+        required=True,
+        default=lambda self: self._get_posting_regime(),
+        help="Removal Entry Policy \n"
+             "  * Residual Value: The non-depreciated value will be "
+             "posted on the 'Residual Value Account' \n"
+             "  * Gain/Loss on Sale: The Gain or Loss will be posted on "
+             "the 'Plus-Value Account' or 'Min-Value Account' ")
+    note = fields.Text('Notes')
+
+    @api.model
+    def _default_sale_value(self):
+        return self._get_sale()['sale_value']
+
+    @api.model
+    def _default_account_sale_id(self):
+        return self._get_sale()['account_sale_id']
+
+    def _get_sale(self):
+        asset_id = self._context.get('active_id')
+        sale_value = 0.0
+        account_sale_id = False
+        inv_lines = self.env['account.invoice.line'].search(
+            [('asset_id', '=', asset_id)])
+        for line in inv_lines:
+            inv = line.invoice_id
+            comp_curr = inv.company_id.currency_id
+            inv_curr = inv.currency_id
+            if line.invoice_id.state in ['open', 'paid']:
+                account_sale_id = line.account_id.id
+                amount = line.price_subtotal
+                if inv_curr != comp_curr:
+                    amount = comp_curr.compute(amount)
+                sale_value += amount
+        return {'sale_value': sale_value, 'account_sale_id': account_sale_id}
+
+    @api.model
+    def _default_account_plus_value_id(self):
+        asset_id = self._context.get('active_id')
```

Comment:
use a many2one and pass `default_asset_id = self.id` when you return the action
then you can remove all the `browse` call